File: 3335-minimum-operations-to-write-the-letter-y-on-a-grid/3335-minimum-operations-to-write-the-letter-y-on-a-grid.py
class Solution:
    def minimumOperationsToWriteY(self, grid: list[list[int]]) -> int:
        n = len(grid)
        c = n // 2
        cntY, cntN = [0, 0, 0], [0, 0, 0]
        totY, totN = 0, 0
        for i in range(n):
            for j in range(n):
                isY = (i == j and i <= c) or (i + j == n - 1 and i <= c) or (j == c and i >= c)
                if isY:
                    cntY[grid[i][j]] += 1
                    totY += 1
                else:
                    cntN[grid[i][j]] += 1
                    totN += 1
        ans = float('inf')
        for a in range(3):
            for b in range(3):
                if a == b:
                    continue
                ans = min(ans, (totY - cntY[a]) + (totN - cntN[b]))
        return ans

File: 3335-minimum-operations-to-write-the-letter-y-on-a-grid/3335-minimum-operations-to-write-the-letter-y-on-a-grid.cpp
class Solution {
public:
    int minimumOperationsToWriteY(vector<vector<int>>& grid) {
        int totY = 0;
        int notY = 0;
        vector<int> yCounts = {0, 0, 0};
        vector<int> notYCounts = {0, 0, 0};
        int n = grid.size();
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < n; ++j) {
                bool isY = ((i <= n / 2) && ((i == j) || (i + j == n - 1))) || (i > n / 2 && j == n / 2);
                if (isY) {
                    ++yCounts[grid[i][j]];
                    ++totY;
                } else {
                    ++notYCounts[grid[i][j]];
                    ++notY;
                }
            }
        }

        int min = std::numeric_limits<int>::max();
        for (int i = 0; i < 3; ++i) { 
            for (int j = 0; j < 3; ++j) {
                if (i == j) continue;
                int changeY = totY - yCounts[i];
                int changeNotY = notY - notYCounts[j];
                min = std::min(min, changeY + changeNotY);
            } 
        }

        return min;
    }
};